src/main/java/com/ecommerce/galleon/controller/WishListController.java

```diff
@@ -0,0 +1,74 @@
+package com.ecommerce.galleon.controller;
+
+
+import com.ecommerce.galleon.common.ApiResponse;
+import com.ecommerce.galleon.dto.ProductDto;
+import com.ecommerce.galleon.model.Product;
+import com.ecommerce.galleon.model.User;
+import com.ecommerce.galleon.model.WishList;
+import com.ecommerce.galleon.service.AuthenticationService;
+import com.ecommerce.galleon.service.WishListService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/wishlist")
+public class WishListController {
+
+    @Autowired
+    WishListService wishListService;
+
+    @Autowired
+    AuthenticationService authenticationService;
+
+
+    // save product as wishlist item
+    @PostMapping("/add")
+    public ResponseEntity<ApiResponse> addToWishList(@RequestBody Product product,
+                                                     @RequestParam("token") String token) {
+        // authenticate the token
+        authenticationService.authenticate(token);
+
+
+        // find the user
+
+        User user = authenticationService.getUser(token);
+
+        // save the item in wishlist
+
+        WishList wishList = new WishList(user, product);
+
+        wishListService.createWishlist(wishList);
+
+        ApiResponse apiResponse = new ApiResponse(true, "Added to wishlist");
```

Comment:
It's better to have a resource file for texts. may we need to handle texts in other languages in future.